lib/fs.js

```diff
@@ -732,21 +742,33 @@ fs.truncateSync = function(path, len) {
 };
 
 fs.ftruncate = function(fd, len, callback) {
+  if (!Number.isInteger(fd) || fd < 0)
+    throw new TypeError('fd must be a valid file descriptor');
+
   if (typeof len === 'function') {
     callback = len;
     len = 0;
-  } else if (len === undefined) {
+  } else if (len === undefined || len === null) {
     len = 0;
+  } else if (!Number.isInteger(len) || len < 0) {
+    throw new TypeError('length must be a positive integer');
   }
+
   var req = new FSReqWrap();
   req.oncomplete = makeCallback(callback);
   binding.ftruncate(fd, len, req);
 };
 
 fs.ftruncateSync = function(fd, len) {
-  if (len === undefined) {
+  if (!Number.isInteger(fd) || fd < 0)
```

Comment:
On a similar note, mind adding the change in native where fd's are translated using `Int32Value()` to `Uint32Value()`?